modules/openapi-generator/src/main/resources/python/api_client.mustache

```diff
@@ -318,7 +316,10 @@ class ApiClient:
                 if content_type is not None:
                     match = re.search(r"charset=([a-zA-Z\-\d]+)[\s;]?", content_type)
                 encoding = match.group(1) if match else "utf-8"
-                response_text = response_data.data.decode(encoding)
+                if response_data.data is None: # should not happen
+                    response_text = ""
+                else:
+                    response_text = response_data.data.decode(encoding)
```

Comment:
To be safe and more explicit, I would assert here:

```suggestion
                msg = "RESTResponse.read() must be called before passing it to response_deserialize()"
                assert response.data is not None, msg
                response_text = response_data.data.decode(encoding)
```

Before, `.decode()` was called on `response_data.data`, which would have failed had it ever been `None`. So the assert will not introduce any extra failures.

Reply:
The assert actually needs to be at the start of the function, to fix typing of `ApiResponse.raw_data`.